### PR TITLE
Refactor/minimum currency amounts

### PR DESCRIPTION
**Pull request by chrisfromthelc**
Related to https://github.com/Automattic/gold/issues/91

## Proposed Changes

* This change will pull the minimum currency amounts from the Memberships library on WPCOM rather than store it in Calypso, negating the need to update in multiple locations should those amounts change in the future.

## Testing Instructions

1. Switch to this branch.
2. Additionally, patch D118035-code
3. On a site with an appropriate plan, go to Tools > Earn, then click Manage Payment Button.
4. Click to add a new payment plan.
5. Ensure that the currency list loads, and if you attempt to use a lower amount than listed for the currency in `class.membership-product.php`, you should get an error. 
6. Attempt to add a plan at the minimum for the currency; it should create the plan as expected.

Screen Shot 2023-08-27 at 2 16 18 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx [hunk 1]
```diff
@@ -40,46 +43,25 @@ type DefaultNames = {
...
  * @returns {number} Minimum transaction amount for given currency.
  */
 function minimumCurrencyTransactionAmount(
+	currency_min: StripeMinimumCurrencyAmounts,
 	currency: string,
 	connectedAccountDefaultCurrency: string
 ): number {
-	if ( connectedAccountDefaultCurrency === currency.toUpperCase() ) {
-		return STRIPE_MINIMUM_CURRENCY_AMOUNT[ currency ];
+	if ( connectedAccountDefaultCurrency.toUpperCase() === currency ) {
```
Comment (millerf) [inline client/my-sites/earn/components/add-edit-plan-modal/index.tsx:61]:
```suggestion
	if ( connectedAccountDefaultCurrency === currency ) {
```

Seing the old check,. I don't think the toUpperCase is needed here? 🤔

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx [hunk 2]
```diff
@@ -106,8 +91,10 @@ const RecurringPaymentsPlanAddEditModal = ( {
 		product?.buyer_can_change_amount ?? false
 	);
 
+	const currencyList = Object.keys( connectedAccountMinimumCurrency );
+
 	const defaultCurrency = useMemo( () => {
-		const flatCurrencyList = currencyList.map( ( e ) => e.code );
+		const flatCurrencyList = currencyList.map( ( e ) => e.toUpperCase() );
```
Comment (millerf) [inline client/my-sites/earn/components/add-edit-plan-modal/index.tsx]: Same here...  [They seem to be](https://github.com/Automattic/jetpack/blob/55861aaa63d4cde01974afc51b03d0a6ed03075b/projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php#L107) in uppercase already... And `currencyList` is already a flat list...

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx [hunk 3]
```diff
@@ -334,7 +331,7 @@ const RecurringPaymentsPlanAddEditModal = ( {
 							onChange={ handlePriceChange }
 							currencySymbolPrefix={ currentCurrency }
 							onCurrencyChange={ handleCurrencyChange }
-							currencyList={ currencyList }
+							currencyList={ currencyList.map( ( code ) => ( { code } ) ) }
```
Comment (millerf) [inline client/my-sites/earn/components/add-edit-plan-modal/index.tsx:333]: Nicely found.

**Review by millerf — APPROVED** [review]
It looks like it is working fine... I added a couple of comments regarding some probably unecessary check

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx (hunk 1 above)
Comment (chrisfromthelc) [reply client/my-sites/earn/components/add-edit-plan-modal/index.tsx:61]: I logged these out before the `if` check without any modifiers. It seems that `connectedAccountDefaultCurrency` does need to be uppercased.
Screen Shot 2023-09-03 at 11 35 18 PM

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx (hunk 2 above)
Comment (chrisfromthelc) [reply client/my-sites/earn/components/add-edit-plan-modal/index.tsx]: Fixed.

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx [hunk 4]
```diff
@@ -94,7 +94,7 @@ const RecurringPaymentsPlanAddEditModal = ( {
 	const currencyList = Object.keys( connectedAccountMinimumCurrency );
 
 	const defaultCurrency = useMemo( () => {
-		const flatCurrencyList = currencyList.map( ( e ) => e.toUpperCase() );
+		const flatCurrencyList = currencyList;
```
Comment (millerf) [inline client/my-sites/earn/components/add-edit-plan-modal/index.tsx]: Do we really need this, then?

Unecessary variable allocation make the code longer for no reason and can be confusing when working again on the code...

 Simply using `currencyList` here would be fine...

**Review by millerf — DISMISSED** [review]

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx (hunk 4 above)
Comment (chrisfromthelc) [reply client/my-sites/earn/components/add-edit-plan-modal/index.tsx]: I've removed this and updated where `flatCurrency` was used.

**Pull request merged**